File: position-update-service/src/main/java/com/ironhack/positionupdateservice/model/PositionUpdate.java
package com.ironhack.positionupdateservice.model;

import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
public class PositionUpdate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @CreationTimestamp
    private LocalDateTime timestamp;
    @Column(precision = 20, scale = 8, columnDefinition="DECIMAL(20,8)")
    private BigDecimal amount;
    private String description;
    private Long idPosition;

    public PositionUpdate() {
    }

    public PositionUpdate(LocalDateTime timestamp, BigDecimal amount, String description, Long idPosition) {
        this.timestamp = timestamp;
        this.amount = amount;
        this.description = description;
        this.idPosition = idPosition;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Long getIdPosition() {
        return idPosition;
    }

    public void setIdPosition(Long idPosition) {
        this.idPosition = idPosition;
    }
}

File: frontend/src/app/models/position-update.ts
export class PositionUpdate {

    // private BigDecimal amount;
    // private String description;
    // private LocalDateTime timestamp;
    constructor(
        private _id: number,
        private _amount: number,
        private _description: string, 
        private _timestamp: string,
    ) {}

    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    
    public get amount(): number {
        return this._amount;
    }

    public get description(): string {
        return this._description
    }

    public get timestamp(): string {
        return this._timestamp;
    }

    public set amount(amount: number) {
        this._amount = amount;
    }

    public set description(description: string) {
        this._description = description;
    }

    public set timestamp(timestamp: string) {
        this._timestamp = timestamp;
    }
}
